solutions/usaco-790.mdx

```diff
@@ -2,14 +2,90 @@
 id: usaco-790
 source: USACO Gold 2018 January
 title: Cow At Large
-author: Qi Wang
+author: Qi Wang, Jesse Choe
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_atlarge_gold_jan18.html)
 
 **Time Complexity:** $\mathcal{O}(N)$
 
+Suppose that a given farmer reaches Bessie at barn $i$; this is only possible if that farmer reached barn $i$ either: 
+
+* At the same time as Bessie
+* At an earlier time than Bessie
+
+This means that we can treat this problem as a shortest path problem, where the farmer reaches barn $i$ at the same time or earlier than Bessie if $\texttt{dist1[i]} \leq \texttt{dist2[i]}$. Here, $\texttt{dist1[i]}$ represents the shortest distance between any given farmer and the $i$-th barn, and $\texttt{dist2[i]}$ represents the shortest distance between Bessie and the $i$-th barn.
```

Comment:
i mean the graph is a tree right? so wouldn't it just be the depth of the tree?
not saying the shortest path analogy is invalid tho

Reply:
Nah that works, but I thought of this like a SP problem.

Reply:
Also, the Java solution uses depth of tree, so I wanted to change it up a bit. If u want me to use tree depth instead of dist1, i'll do it, but I won't change my code lol.